### ISSUE TITLE
Editor has Case-sensitive rules for file-extensions to show FileViewPanel

### ISSUE THREAD

**Issue by phomm** [castle-engine/castle-engine#611]
Working under windows, didn't test other OSes.
here is sample test:
<img width="185" alt="image" src="https://github.com/castle-engine/castle-engine/assets/2115136/c151b38e-8062-492b-b504-1c5f416acdf4">
<img width="171" alt="image" src="https://github.com/castle-engine/castle-engine/assets/2115136/f4520d50-4ecf-4f44-bb11-100727b36de7">
<img width="278" alt="image" src="https://github.com/castle-engine/castle-engine/assets/2115136/10dad563-f8d4-472c-8785-3727ab287ab5">
<img width="279" alt="image" src="https://github.com/castle-engine/castle-engine/assets/2115136/e07e0c9e-cc8f-413f-9a71-8e0dc122a116">
<img width="281" alt="image" src="https://github.com/castle-engine/castle-engine/assets/2115136/b8cb907d-42bd-4599-a03a-c5627f053327">
<img width="183" alt="image" src="https://github.com/castle-engine/castle-engine/assets/2115136/57bc9874-36f4-4f70-bedd-73debd3eda46">
Last 2 images are same, except for filename with 1 and extension case

Checked the code, seems that (a bit strange) conditional compilation decision in this definition is causing it 
https://github.com/castle-engine/castle-engine/blob/6eb87225cb141a4421f46c8e3d465f14a1d96077/src/base/castleutils_filenames.inc#L28
while used in 
https://github.com/castle-engine/castle-engine/blob/6eb87225cb141a4421f46c8e3d465f14a1d96077/src/files/castlefilefilters.pas#L215

**Comment by phomm** [comment]
Yes, it should ignore case by default on windows, but bc it was build under FPC with this (I named it strange) conditional directive, it bypasses this rule. So I emphasize this particular line. Maybe it is ok to remove {$ifndefFPC} ?

**Comment by michaliskambi** [comment]
Ah! I think I actually see the problem in this case. I was just sleepy and forgot `not` in this :) The 3rd parameter to `IsWild` should not `not FileNameCaseSensitive` to do what I wanted :)

**Comment by michaliskambi** [comment]
So this is now explained and fixed:

> Moreover it's weird you experience this bug on Windows where we should ignore case by default, not honor it :)

This was just a trivial bug :)

This is still TODO, but it only really matters now for users of non-Windows filesystems (like Linux) that use non-lowercase extensions (like `myimage.PNG`)

> And indeed, for decisions like "does this match a file extension" -- these should always ignore case, on all systems. Linux file manages also generally ignore case for this, e.g. I can open foo.PNG on Linux, as image, without any problems.
